library/management/chat/src/main/java/org/quiltmc/qsl/chat/impl/InternalQuiltChatApiUtil.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.chat.impl;
+
+import org.quiltmc.qsl.chat.api.QuiltMessageType;
+
+import java.util.EnumSet;
+
+/**
+ * Provides some shortcuts for common code in the chat api
+ */
+public final class InternalQuiltChatApiUtil {
+	private InternalQuiltChatApiUtil() {}
+
+	public static EnumSet<QuiltMessageType> s2cType(QuiltMessageType type, boolean isOnClientSide) {
+		if (isOnClientSide) {
+			return EnumSet.of(type, QuiltMessageType.CLIENT, QuiltMessageType.INBOUND);
+		} else {
+			return EnumSet.of(type, QuiltMessageType.SERVER, QuiltMessageType.OUTBOUND);
+		}
```

Comment:
oh dear, i don't like this pattern and yet the only real alternative that i have to offer here are `? :`s, which might be worse

Reply:
What dont you like about it out of curiosity? It seems like pretty normal code to me

Reply:
Excellent question! I don't know! I just don't like it's face! but yeah, compared to other alternatives? yeah, this might be the best one